src/commands/dev/types.d.ts

```diff
@@ -8,23 +8,14 @@ export type BuildConfig = NonNullable<NetlifyTOML['build']>
 export type DevConfig = NonNullable<NetlifyTOML['dev']> & {
   framework: FrameworkNames
   /** Directory of the functions */
-  functions?: string
-  publish?: string
-  /** Port to serve the functions */
-  port: number
-  live: boolean
+  functions?: string | undefined
+  live?: boolean | undefined
   /** The base directory from the [build] section of the configuration file */
-  base?: string
-  staticServerPort?: number
-  functionsPort?: number
-  autoLaunch?: boolean
-  https?: {
-    keyFile: string
-    certFile: string
-  }
-  envFiles?: string[]
+  base?: string | undefined
+  staticServerPort?: number | undefined
+  envFiles?: string[] | undefined
 
-  jwtSecret: string
-  jwtRolePath: string
-  pollingStrategies?: PollingStrategy[]
```

Comment:
This is normalized to a string before it gets here: https://github.com/netlify/build/blob/1b9f2f40de7ded6285e9b7c08a1adb43550a0ccf/packages/build-info/src/settings/get-build-settings.ts#L102